Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Middlewares/Sha256TreeHashMiddleware.swift

```diff
@@ -24,14 +24,8 @@ public struct Sha256TreeHashMiddleware<OperationStackInput, OperationStackOutput
         context: Context
     ) async throws {
         switch request.body {
-        case .data(let data):
-            guard let data = data else {
-                return
-            }
-            if !request.headers.exists(name: X_AMZ_CONTENT_SHA256_HEADER_NAME) {
-                let sha256 = try data.computeSHA256().encodeToHexString()
-                builder.withHeader(name: X_AMZ_CONTENT_SHA256_HEADER_NAME, value: sha256)
-            }
+        case .data:
+            break
```

Comment:
Glacier never uses tree hash on a non-streaming body?

Reply:
It's that we shouldn't be assigning this header ourselves and instead let CRT's signer assign it. As it is now, this results in error because we shouldn't set this header before we hand off the request to CRT's signer.